src/arrayMethodJoin.js

```diff
@@ -1,11 +1,41 @@
 'use strict';
 
+const nullString = 'null';
+const emptyString = '';
 /**
  * Implement method join
  */
+
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+
+    if (this.length === 0) {
+      return emptyString;
+    } else if (this.length === 1) {
+      return this[0].toString();
+    }
```

Comment:
use early return pattern

```suggestion
    if (this.length === 0) {
      return emptyString;
    } 
// By the way, do you really need it? Wouldn't the method work without this condition?
    if (this.length === 1) {
      return this[0].toString();
    }
```